### PR TITLE
feat: Add per-font weight restrictions and auto-detection for uploaded font kits

### LINKED ISSUE #94: Auto-detect available font weights on install

### PR DESCRIPTION
**Pull request by mattcowan**
Closes #94 

- Implemented per-font weight restrictions allowing configuration of available font weights in the admin panel (Settings → Typography Stylist → Custom Fonts).
- Introduced single-weight auto-apply feature, hiding the weight selector when only one weight is available and automatically applying it.
- Implemented weight validation on font change to adjust the weight to the closest available option.
- Updated admin UI to reflect changes, including renaming "Edit Fallbacks" to "Edit Settings" for clarity.
- Fixed CSS variable trailing comma issue that caused fonts without fallbacks to fail loading.
- Improved accessibility and user experience in the inline editor and font management sections.

This pull request introduces a new feature that allows restricting available font weights per font, improving both the admin UI and the block editor experience. The changes ensure that only the correct font weights can be selected for each font, provide utility functions and comprehensive tests for weight filtering, and update the UI to reflect these restrictions. Below are the most significant changes:

**Admin UI and Data Handling:**

* The admin page JavaScript now collects selected available font weights when saving font settings for all font types (custom, Adobe, manual), sending them to the backend as the `available_weights` property. [[1]](diffhunk://#diff-da17bb06f0b71c18c492b2116430a2d5efd2391fd081bd24bf2fd5ae249ec3b2R658-R669) [[2]](diffhunk://#diff-da17bb06f0b71c18c492b2116430a2d5efd2391fd081bd24bf2fd5ae249ec3b2R722-R733) [[3]](diffhunk://#diff-da17bb06f0b71c18c492b2116430a2d5efd2391fd081bd24bf2fd5ae249ec3b2R784-R787) [[4]](diffhunk://#diff-da17bb06f0b71c18c492b2116430a2d5efd2391fd081bd24bf2fd5ae249ec3b2L794-R807)
* New CSS classes were added to style the font weight checkboxes in the admin interface, improving usability for selecting available weights.

**Block Editor Logic and UI:**

* The block editor's internal `fontIdMap` now stores an `availableWeights` array for each font, populated from the backend, allowing the editor to know which weights are valid for each font. [[1]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79L1516-R1530) [[2]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79L1533-R1547) [[3]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79L1544-R1558) [[4]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79L1560-R1632) [[5]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL1011-R1011) [[6]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL1026-R1026) [[7]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL1037-R1037)
* When a user selects a font in the block editor, the editor now checks if the current weight is valid for that font. If not, it automatically selects the closest available weight, or the only available weight if there is just one.
* The font weight dropdown in the block editor is dynamically filtered to only show the allowed weights for the selected font. If a font has only one available weight, the dropdown is hidden.

**Utility Functions and Testing:**

* Utility functions `getFilteredWeightOptions` and `getClosestWeight` were added to filter weight options and find the closest valid weight, respectively. These are used in both the block editor and are exposed for unit testing. [[1]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79L1560-R1632) [[2]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL32-R32)
* A comprehensive test suite was added to validate the behavior of weight filtering and closest weight selection logic, ensuring robust handling of all edge cases.

### CODE REVIEW

**Comment by mattcowan** [comment]
Taking auto-detection out...will resubmit.

File: includes/admin-page.php [hunk 1]
```diff
@@ -20,6 +20,53 @@
...
+        '800' => __('800 - Extra Bold', 'typography-stylist'),
+        '900' => __('900 - Black', 'typography-stylist'),
+    );
+    $available = !empty($font['available_weights']) ? $font['available_weights'] : array();
+    $all_available = empty($available); // Empty array = all weights available
+    ?>
+    <div class="typost-form-field">
+        <label><?php esc_html_e('Available Font Weights:', 'typography-stylist'); ?></label>
+        <p class="description" id="typost-<?php echo esc_attr($prefix); ?>-weights-desc-<?php echo esc_attr($font['id']); ?>">
+            <?php esc_html_e('Uncheck weights this font doesn\'t include to exclude them from being selected. You can leave all checked for variable fonts.', 'typography-stylist'); ?>
```
[the remark this answers is not in the document]
Comment (mattcowan) [reply includes/admin-page.php:48]: This is a standard escape sequence.

File: typography-stylist.php [hunk 2]
```diff
@@ -3364,7 +3374,7 @@ public function update_manual_font_endpoint($request) {
         $id = sanitize_key($request->get_param('id'));
         $params = $request->get_json_params();
 
-        if (!isset($params['font_family'])) {
+        if (!isset($params['font_family']) && !isset($params['available_weights'])) {
             return new WP_Error('missing_font_family', esc_html__('Font family parameter is required', 'typography-stylist'), array('status' => 400));
```
[the remark this answers is not in the document]
Comment (mattcowan) [reply typography-stylist.php]: Updated messages in [906685c](https://github.com/mattcowan/typography-stylist-for-wordpress/pull/116/commits/906685c9b47b4c47a7503e0568de2b40a4326065)

File: typography-stylist.php [hunk 3]
```diff
@@ -3199,7 +3204,7 @@ public function update_font_fallback_endpoint($request) {
         $id = sanitize_key($request->get_param('id'));
         $params = $request->get_json_params();
 
-        if (!isset($params['fallbacks'])) {
+        if (!isset($params['fallbacks']) && !isset($params['available_weights'])) {
             return new WP_Error('missing_fallbacks', esc_html__('Fallbacks parameter is required', 'typography-stylist'), array('status' => 400));
```
[the remark this answers is not in the document]
Comment (mattcowan) [reply typography-stylist.php]: Updated messages in [906685c](https://github.com/mattcowan/typography-stylist-for-wordpress/pull/116/commits/906685c9b47b4c47a7503e0568de2b40a4326065)

File: typography-stylist.php [hunk 4]
```diff
@@ -3121,7 +3121,7 @@ public function update_adobe_font_fallback_endpoint($request) {
...
```
[the remark this answers is not in the document]
Comment (mattcowan) [reply typography-stylist.php]: Updated messages in [906685c](https://github.com/mattcowan/typography-stylist-for-wordpress/pull/116/commits/906685c9b47b4c47a7503e0568de2b40a4326065)
